builder/setup_linux.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package builder
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/Azure/acr-builder/util"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+const (
+	config = `{
+		"experimental": "enabled",
+		"HttpHeaders": {"X-Meta-Source-Client": "ACR-BUILDER"}
+	}`
+)
+
+// setupConfig initializes ~/.docker/config.json
+func (b *Builder) setupConfig(ctx context.Context) error {
+	args := []string{
+		"docker",
+		"run",
+		"--name", fmt.Sprintf("acb_init_config_%s", uuid.New()),
+		"--rm",
+
+		// Home
+		"--volume", homeVol + ":" + homeWorkDir,
+		"--env", "HOME=" + homeWorkDir,
+		"--volume", util.GetDockerSock(),
+		"--entrypoint", "bash",
+		"ubuntu",
```

Comment:
>ubuntu [](start = 3, length = 6)

later we probably can try busybox =)

Reply:
Yeah definitely. It's a lot more significant on Windows, I just chose two that we have cached right now.